src/bindgen/ir/ty.rs

```diff
@@ -306,6 +306,24 @@ impl PrimitiveType {
     fn can_cmp_eq(&self) -> bool {
         true
     }
+
+    pub(crate) fn can_be_enum_fixed_type(&self) -> bool {
```

Comment:
I think I'd remove this. We don't check it for other cases, and as you said it's invalid rust so there's little point in checking it.